oeps/oep-0008.rst

```diff
@@ -0,0 +1,372 @@
+  to be installed for different Python virtual environments.
+
+* Requirements files are kept either at root level or within a directory named
+  ``requirements``.
+
+* Direct dependencies are listed in ``requirements/*.in`` for use in
+  development and `Tox`_ virtual environments with as few version constraints
+  as possible.
+
+* Full pinned dependencies in ``requirements/*.txt`` are auto-generated via
+  ``make pip-compile``.
+
+* Requirements filename examples:
+
+  * ``requirements.*``: contains the base requirements
+
+  * ``test-requirements.*``: contains the requirements needed for writing and
+    running tests
+
+* Editable requirements format example:
+  ``-e git+https://github.com/edx/django-oauth2-provider.git@0.2.7-fork-edx-6a#egg =django-oauth2-provider==0.2.7-fork-edx-6``
+
+Code Quality
+^^^^^^^^^^^^
+
+* `Pylint`_ for static code analysis for conformance with Python best
+  practices
+
+  * `edx-lint`_ for generating pylint configuration files that verify edX best
+    practices
+  * Check both the ``pylintrc`` and the ``pylintrc_tweaks`` files into the
+    repo
+
+* `pycodestyle`_ (formerly pep8) for checking compliance with style
+  conventions in PEP 8
+
+  * Configuration file: ``[pycodestyle]`` section in ``setup.cfg``
+
+* `pydocstyle`_ for checking compliance with Python docstring conventions
+
+  * Configuration file: ``[pydocstyle]`` section in ``setup.cfg``
+
+Test Framework
+^^^^^^^^^^^^^^
+
+* `pytest`_ as the unit test framework
+
+  * Configuration file: ``pytest.ini``
+
+* Tox for installing packages and running tests under multiple Python
+  versions and interpreters
+
+  * Configuration file: ``tox.ini``
+
+Code Coverage
+^^^^^^^^^^^^^
+
+* `Coverage.py`_ for calculating code coverage
+
+  * Configuration file: ``.coveragerc``
+
+* Report coverage stats to `codecov`_ for trending and analysis
+
+.. _pip-tools: https://pypi.python.org/pypi/pip-tools
+.. _Pylint: https://www.pylint.org/
+.. _edx-lint: https://github.com/edx/edx-lint/
+.. _pycodestyle: http://pycodestyle.pycqa.org/
+.. _pydocstyle: http://www.pydocstyle.org/
+.. _pytest: http://doc.pytest.org/
+.. _Tox: https://tox.readthedocs.io/
+.. _Coverage.py: https://coverage.readthedocs.io/
+.. _codecov: https://codecov.io/
+
+
+JavaScript Language Specific
+----------------------------
+
+Code Quality
+^^^^^^^^^^^^
+
+* `ESLint`_ for static analysis in order to find problematic patterns or code
+  that doesn’t adhere to style guidelines
+
+  * `edX ESLint config`_  for generating ESLint configuration files that verify
+    edX best practices
+
+  * Configure in either ``package.json`` or ``.eslintrc.json``
+
+  * Specify exclusions in ``.eslintignore``
+
+
+Test Framework
+^^^^^^^^^^^^^^
+
+* `Jasmine`_ as the framework for testing JavaScript code
+
+  * `jasmine-jquery`_ matchers and fixture loader for Jasmine
+
+  * Jasmine-based helper classes from the testing folder of the `edX UI
+    Toolkit`_
+
+* `Karma`_ as the test runner for the JS tests
```

Comment:
Karma's a pretty configurable tool, it may be worth going into some of what we expect to get out of it here, eg: should tests run in the terminal (node) or in the browser? If so, what browsers? Multiple in parallel? If not, how do we approach debugging? Should we mock browser APIs? We have a few different approaches across different projects.